contracts/examples/ExampleSlidingWindowOracle.sol

```diff
@@ -99,26 +103,24 @@ contract ExampleSlidingWindowOracle {
 
     // returns the amount out corresponding to the amount in for a given token using the moving average over the time
     // range [now - windowSize, now]
-    // update must have been called for the bucket corresponding to `now - period` as well as the bucket corresponding
-    // to `now`
+    // update must have been called for the bucket corresponding to timestamp `now - period` as well as the bucket
+    // corresponding to `now`
     function consult(address tokenIn, uint amountIn, address tokenOut) external view returns (uint amountOut) {
         address pair = UniswapV2Library.pairFor(factory, tokenIn, tokenOut);
-        Observation storage historicalObservation = getHistoricalObservation(pair);
-        (
-            uint32 blockTimestamp,
-            uint price0Cumulative,
-            uint price1Cumulative
-        ) = UniswapV2OracleLibrary.currentCumulativePrices(pair);
-
-        // this condition may incorrectly trigger for one period around the uint32(-1) overflow of blockTimestamp
-        uint32 timeElapsed = blockTimestamp - historicalObservation.blockTimestamp;
+        Observation storage firstObservation = getFirstObservationInWindow(pair);
+
+        uint timeElapsed = block.timestamp - firstObservation.timestamp;
         require(timeElapsed <= windowSize, 'SlidingWindowOracle: MISSING_HISTORICAL_OBSERVATION');
+        // should never happen.
+        require(timeElapsed >= windowSize - periodSize * 2, 'SlidingWindowOracle: UNEXPECTED_TIME_ELAPSED');
```

Comment:
probably should be an assert

Reply:
I thought about that, but if it ever did happen a random revert would be pretty terrible

Reply:
hmm i guess since it's an example it's ok